uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/modal/modal-document_test.js

```diff
@@ -198,6 +204,57 @@ define([
       });
     });
 
+    describe('$scope.onContactChanged', function () {
+      beforeEach(function () {
+        spyOn(ContactService, 'updateCache').and.returnValue({});
+        spyOn(AssignmentService, 'search').and.returnValue($q.resolve([]));
+      });
+
+      beforeEach(function () {
+        initController();
+        $scope.onContactChanged(contactMock.contact);
+      });
+
+      it('calls contact service to cache selected Contact', function () {
+        expect(ContactService.updateCache).toHaveBeenCalled();
+      });
+    });
+
+    describe('$scope.onContactChanged', function () {
+      beforeEach(function () {
+        spyOn(ContactService, 'updateCache').and.returnValue({});
+        spyOn(AssignmentService, 'search').and.returnValue($q.resolve([]));
+      });
+
+      beforeEach(function () {
+        initController();
+        $scope.onContactChanged(contactMock.contact);
+      });
+
+      it('resets the document case id to empty', function () {
+        expect($scope.document.case_id).toEqual('');
+      });
+
+      it('calls contact service to cache selected Contact', function () {
+        expect(ContactService.updateCache).toHaveBeenCalled();
+      });
+    });
+
+    describe('$scope.cacheTargetContactAssignments', function () {
+      beforeEach(function () {
+        spyOn(AssignmentService, 'search').and.returnValue($q.resolve([]));
+      });
+
+      beforeEach(function () {
+        initController();
+        $scope.cacheTargetContactAssignments({ id: '204'});
+      });
+
+      it("calls assignment service to search assignments of target contact", function () {
+        expect(AssignmentService.search).toHaveBeenCalledWith(null, null);
```

Comment:
Original function searches through returned assignments and filters based on contract. That is not being tested here.